test/e2e/appsody_knative.go

```diff
@@ -0,0 +1,71 @@
+package e2e
+
+import (
+	goctx "context"
+	"testing"
+	"time"
+
+	"github.com/appsody/appsody-operator/test/util"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	e2eutil "github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	dynclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AppsodyKNativeTest verify functionality of kNative option in appsody
+func AppsodyKnativeTest(t *testing.T) {
+	ctx, err := util.InitializeContext(t, cleanupTimeout, retryInterval)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Cleanup()
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatalf("Couldn't get namespace: %v", err)
+	}
+
+	t.Logf("Namespace: %s", namespace)
+
+	f := framework.Global
+
+	if isKnativeInstalled(t, f) {
```

Comment:
If you reverse your `if` logic ("early exit"), I think it'd make your code more readable and as the test gets more complex with more nested `if-else` it'd be harder to read:

```suggestion
	if !isKnativeInstalled(t, f) {
		t.Log("Knative is not installed on this cluster, skipping AppsodyKnativeTest...")
		return
	}
	...
```